src/app/Shared/Services/NotificationChannel.service.tsx

```diff
@@ -82,40 +145,29 @@ export class NotificationChannel {
     private readonly notifications: Notifications,
     private readonly login: LoginService
   ) {
-    this.messages(NotificationCategory.WsClientActivity).subscribe(v => {
-      const addr = Object.keys(v.message)[0];
-      const status = v.message[addr];
-      notifications.info('WebSocket Client Activity', `Client at ${addr} ${status}`, NotificationCategory.WsClientActivity);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingCreated).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Created', `${event.recording} created in target: ${event.target}`);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingStopped).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Stopped', `${event.recording} was stopped`);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingSaved).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Archived', `${event.recording.name} was archived`);
-    });
-
-    this.messages(NotificationCategory.ActiveRecordingDeleted).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Deleted', `${event.recording} was deleted`);
-    });
-
-    this.messages(NotificationCategory.ArchivedRecordingCreated).subscribe(v => {
-      const event: RecordingNotificationEvent = v.message;
-      notifications.success('Recording Archived', `${event.recording.name} was uploaded into archives`);
+    messageKeys.forEach((value, key) => {
+      if (!value) {
+        return;
+      }
+      this.messages(key).subscribe((msg: NotificationMessage) => {
+        const message = value.body(msg);
+        notifications.notify({ title: value.title, message, category: key, variant: value.variant })
+      });
     });
+    this._messages.pipe(
+      filter(msg => !messageKeys.has(msg.meta.category as NotificationCategory))
```

Comment:
Does this filter only allow notifications that have an invalid NotificationCategory? If it does, what kinds of notifications would meet the conditions on lines 163 and 165?

Reply:
It filters notifications that aren't being handled by any of the defined messageKey "mappers". Those could still have known categories, if for some reason a mapper just wasn't provided. It is a little redundant since we do also have a defined mapper for those two categories on 163/165 already - but then again, we also have defined mappers for *every* category right now, so in practice we never expect this filter to actually come up with any results.